### PR TITLE
fix(offer): improve error message when updating offers with removed endpoints

### PR DESCRIPTION
**Pull request by nvinuesa**
Juju offer is an upsert operation but this might misleading to users. Since we are not changing this behavior right now, this patch at least improves the error message when *updating* an offer and in this update an endpoint (with active consumers) is removed.
The new error message should give the user the hint that they are actually updating the offer and therefore if they don't want to remove the previously added endpoints, they should be added as a list in the passed arguments. This behavior is already documented.

## Checklist

- [X] Code style: imports ordered, good names, simple structure, etc
- [ ] ~Comments saying why design decisions were made~
- [X] Go unit tests, with comments saying what you're testing
- [ ] ~[Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- [ ] ~[doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Bootstrap and add two models; one with influxdb on it and other with grafana:

```
juju bootstrap localhost c
juju add-model m1
juju deploy grafana
juju add-model m2
juju deploy influxdb
```
Offer and integrate grafana and influxdb:
```
juju offer influxdb:grafana-source
juju switch m1
juju integrate grafana admin/m2.influxdb
```
Now, try to update the offer and the correct error should be displayed:
```
juju switch m2
juju offer influxdb:query
ERROR cannot update application offer "influxdb": updating offer influxdb:query would remove endpoint "grafana-source" which has active consumers
```

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2048870

**Jira card:** JUJU-6124

### CODE REVIEW

File: state/applicationoffers.go [hunk 1]
```diff
@@ -705,8 +705,8 @@ func (s *applicationOffers) UpdateOffer(offerArgs crossmodel.AddApplicationOffer
 	return s.makeApplicationOffer(doc)
 }
 
-func (s *applicationOffers) ensureEndpointsNotInUse(appName, offerUUID string, endpoints set.Strings) error {
-	if len(endpoints) == 0 {
+func (s *applicationOffers) ensureEndpointsNotInUse(appName, offerUUID string, removedEndpoints, updatedEndpoitns set.Strings) error {
```
Comment (SimonRichardson) [inline state/applicationoffers.go:708]:
```suggestion
func (s *applicationOffers) ensureEndpointsNotInUse(appName, offerUUID string, removedEndpoints, updatedEndpoints set.Strings) error {
```

Comment (nvinuesa) [reply state/applicationoffers.go]: fixed!

**Review by SimonRichardson — COMMENTED** [review]
Can you make a test with multiple updated endpoints? The error message wouldn't work with multiple ones, or maybe confusing.

**Comment by nvinuesa** [comment]
@SimonRichardson added a test with multiple updated endpoints. It adds some other changes because I had to add another endpoint to the mysql charm.

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**